bundles/org.openhab.binding.publictransportswitzerland/src/main/java/org/openhab/binding/publictransportswitzerland/internal/stationboard/PublicTransportSwitzerlandStationboardHandler.java

```diff
@@ -0,0 +1,252 @@
+ * sent to one of the channels.
+ *
+ * @author Jeremy Stucki - Initial contribution
+ */
+@NonNullByDefault
+public class PublicTransportSwitzerlandStationboardHandler extends BaseThingHandler {
+
+    // Limit the API response to the necessary fields
+    private static final String FIELD_FILTERS = createFilterForFields(
+            "stationboard/to",
+            "stationboard/category",
+            "stationboard/number",
+            "stationboard/stop/departureTimestamp",
+            "stationboard/stop/delay",
+            "stationboard/stop/platform");
+
+    private final ChannelGroupUID dynamicChannelGroupUID = new ChannelGroupUID(getThing().getUID(), "departures");
+
+    private final Logger logger = LoggerFactory.getLogger(PublicTransportSwitzerlandStationboardHandler.class);
+
+    private @Nullable ScheduledFuture<?> updateChannelsJob;
+    private @Nullable ExpiringCache<@Nullable JsonElement> cache;
+
+    public PublicTransportSwitzerlandStationboardHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateChannels();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        PublicTransportSwitzerlandStationboardConfiguration config = getConfigAs(PublicTransportSwitzerlandStationboardConfiguration.class);
+
+        // Together with the 10 second timeout, this should be less than a minute
+        cache = new ExpiringCache<>(45_000, this::updateData);
+
+        if (config.station == null || config.station.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        } else {
+            updateStatus(ThingStatus.UNKNOWN);
+            startChannelUpdate();
+        }
+    }
+
+    @Override
+    public void dispose() {
+        stopChannelUpdate();
+    }
+
+    private void startChannelUpdate() {
+        updateChannelsJob = scheduler.scheduleWithFixedDelay(this::updateChannels, 0, 60, TimeUnit.SECONDS);
+    }
+
+    private void stopChannelUpdate() {
+        if (updateChannelsJob != null && !updateChannelsJob.isCancelled()) {
+            updateChannelsJob.cancel(true);
+        }
+    }
+
+    public @Nullable JsonElement updateData() {
+        PublicTransportSwitzerlandStationboardConfiguration config = getConfigAs(PublicTransportSwitzerlandStationboardConfiguration.class);
+
+        try {
+            String escapedStation = URLEncoder.encode(config.station, StandardCharsets.UTF_8.name());
+            String requestUrl = BASE_URL + "stationboard?station=" + escapedStation + FIELD_FILTERS;
+
+            String response = HttpUtil.executeUrl("GET", requestUrl, 10_000);
+            logger.debug("Got response from API: {}", response);
+
+            return new JsonParser().parse(response);
+        } catch (Exception e) {
+            logger.warn("Unable to fetch stationboard data", e);
```

Comment:
This will log a stacktrace e.g. if the HTTP connection fails. Would logging the expection message enough?
```suggestion
            logger.warn("Unable to fetch stationboard data: {}", e.getMessage());
```